### PR TITLE
73 add simple crud for site from sprint 2

### LINKED ISSUE #73: Add simple CRUD for Site
**Site**: AuditEntity

- Name
- Description
- Urls: List<string> (www.x.com, x.com, etc.)
- Pages: List<Page Id> (PageId is Id for Page class)
- DefaultPage: PageId
- Administrator: RoleId

Route pattern: /admin/sites/
- add
- update
- list
- view

# todo:
- [x] UI
  - [x] List
  - [x] Create
  - [x] Update
  - [x] Delete

### PR DESCRIPTION
**Pull request by ParsaGachkar**
- [x] Entity
- [x] Repository
- [x] Service
- [x] Tests
- [x] Commands & Queries
- [x] APIs

### CODE REVIEW

**Comment by ParsaGachkar** [comment]
@pournasserian @MostafaTech review please

File: src/FluentCMS.Application/Sites/AddUrlSiteCommand.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using MediatR;
+
+namespace FluentCMS.Application.Sites;
+
+public class AddUrlSiteCommand : IRequest
```
Comment (MostafaTech) [inline src/FluentCMS.Application/Sites/AddUrlSiteCommand.cs:5]: I think these names are better for this command:
- AddUrlToSiteCommand
- AddSiteUrlCommand

I prefer `AddSiteUrlCommand`

File: src/FluentCMS.Application/Sites/CreateSiteCommand.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using MediatR;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (MostafaTech) [inline src/FluentCMS.Application/Sites/CreateSiteCommand.cs]: please remove all these unused system namespaces

File: src/FluentCMS.Application/Sites/RemoveUrlSiteCommand.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using MediatR;
+
+namespace FluentCMS.Application.Sites;
+
+public class RemoveUrlSiteCommand : IRequest
```
Comment (MostafaTech) [inline src/FluentCMS.Application/Sites/RemoveUrlSiteCommand.cs:5]: I think these names are better for this command:
- RemoveUrlFromSiteCommand
- RemoveSiteUrlCommand

I prefer `RemoveSiteUrlCommand`

File: src/FluentCMS.Application/Sites/AddUrlSiteCommand.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+{
+    public Guid Id { get; set; }
```
Comment (MostafaTech) [inline src/FluentCMS.Application/Sites/AddUrlSiteCommand.cs:7]: Its better to change simple Id to SiteId

File: src/FluentCMS.Application/Sites/RemoveUrlSiteCommand.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+{
+    public Guid Id { get; set; }
```
Comment (MostafaTech) [inline src/FluentCMS.Application/Sites/RemoveUrlSiteCommand.cs:7]: Its better to change simple Id to SiteId

File: src/FluentCMS.Application/Sites/SiteHandlers.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
+﻿using FluentCMS.Entities.Sites;
+using FluentCMS.Services;
+using MediatR;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (MostafaTech) [inline src/FluentCMS.Application/Sites/SiteHandlers.cs]: please remove all these unused system namespaces

File: src/FluentCMS.Domain/Services/SiteService.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FluentCMS.Services;
+public class SiteService(ISiteRepository siteRepository)
+{
+    private readonly ISiteRepository _siteRepository = siteRepository;
```
Comment (MostafaTech) [inline src/FluentCMS.Domain/Services/SiteService.cs:8]: When you are using primary constructors, you dont need to use this field in your class.

**Review by MostafaTech — APPROVED** [review]

**Comment by MostafaTech** [comment]
I think its enough for this PR.
Lets work on the UI in another PR.

**Pull request merged**